src/test/java/io/appium/java_client/android/FingerPrintTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.android;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import io.appium.java_client.remote.MobileCapabilityType;
+import io.appium.java_client.service.local.AppiumDriverLocalService;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+/**
+ * It is necessary to make this test passing
+ * - emulator API Level 23 Nexus device
+ * - perform 'adb emu finger touch' 1 on Windows
+ */
+public class FingerPrintTest {
+
+    private static final String PASSWORD_INPUT_ID = "com.android.settings:id/password_entry";
+    private static final String FIRST_IN_LIST_XPATH = "//android.widget.ListView[1]/android.widget.LinearLayout[1]";
+    private static AppiumDriverLocalService service;
+
+    /**
+     * initialization.
+     */
+    @BeforeClass public static void beforeClass() {
+        service = AppiumDriverLocalService.buildDefaultService();
+        service.start();
+
+        if (service == null || !service.isRunning()) {
```

Comment:
service cannot be `null` here

Reply:
@mykola-mokhnach it was copied from **BaseAndroidTest**  and for sure it can not be null here :)